packages/extension-ui/src/hooks/useGenesisHashOptions.ts

```diff
@@ -28,35 +35,35 @@ export default function (): Option[] {
       .catch(console.error);
   }, []);
 
-  const hashes = useMemo(
-    () => [
-      {
-        text: t('Allow use on any chain'),
-        value: ''
+  const hashes = useMemo(() => {
+    const { alephChains, otherChains, relayChains } = chains.reduce(
+      (acc: ChainsAccumulator, { chain, genesisHash }) => {
+        if (chain.includes(ALEPH_ZERO)) {
+          acc.alephChains.push({ text: chain, value: genesisHash });
+        } else if (chain.includes(RELAY_CHAIN)) {
+          acc.relayChains.push({ text: chain, value: genesisHash });
+        } else {
+          acc.otherChains.push({ text: chain, value: genesisHash });
+        }
+
+        return acc;
       },
-      ...chains
-        .filter(({ chain }) => chain.includes(RELAY_CHAIN))
-        .map(({ chain, genesisHash }) => ({
-          text: chain,
-          value: genesisHash
-        })),
-      ...chains
-        .map(({ chain, genesisHash }) => ({
-          text: chain,
-          value: genesisHash
-        }))
-        // remove the relay chains, they are at the top already
-        .filter(({ text }) => !text.includes(RELAY_CHAIN))
-        .concat(
-          // get any chain present in the metadata and not already part of chains
-          ...metadataChains.filter(({ value }) => {
-            return !chains.find(({ genesisHash }) => genesisHash === value);
-          })
-        )
-        .sort((a, b) => a.text.localeCompare(b.text))
-    ],
-    [metadataChains, t]
-  );
+      { alephChains: [], relayChains: [], otherChains: [] }
+    );
+
+    const newChains = [
+      ...alephChains,
+      { text: t('Allow use on any chain'), value: '' },
+      ...relayChains,
+      ...otherChains.filter(({ text }) => !text.includes(RELAY_CHAIN) && !text.includes(ALEPH_ZERO))
```

Comment:
Doesn't the above `reduce()` make sure that `otherChains` does not include _relay_ and _aleph_ chains, making the filtering redundant?